src/js/common/schemaform/widgets/SSNWidget.jsx

```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import TextWidget from './TextWidget';
+
+/*
+ * Handles removing dashes from SSNs by keeping the user input in local state
+ * and saving the transformed version instead
+ */
+export default class SSNWidget extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { val: props.value };
+  }
+  handleChange = (val) => {
+    const strippedSSN = val.replace(/[\- ]/g, '');
+    this.setState({ val }, () => {
+      this.props.onChange(strippedSSN);
+    });
+  }
+  render() {
+    return <TextWidget type="email" {...this.props} value={this.state.val} onChange={this.handleChange}/>;
```

Comment:
Because I copied and pasted this widget from EmailWidget. That's not a good reason, but it is a reason. :)